$OpenBSD: patch-src_log_c,v 1.1 2020/09/22 14:12:21 danj Exp $

This fixes `haproxy: vfprintf %s NULL in "<<<<>%s %2d %02d:%02d:%02d %.*s%*s"`
in syslog. It was never committed upstream and commit 5464885 shuffled the
code.

It was discussed on the mailing list:
https://www.mail-archive.com/haproxy@formilux.org/msg35601.html

Index: src/log.c
--- src/log.c.orig
+++ src/log.c
@@ -1388,7 +1388,7 @@ static char *update_log_hdr(const time_t time)
 				   "<<<<>%s %2d %02d:%02d:%02d %.*s%*s",
 				   monthname[tm.tm_mon],
 				   tm.tm_mday, tm.tm_hour, tm.tm_min, tm.tm_sec,
-				   (int)host.data, host.area, sep, "");
+				   (int)host.data, host.area ? host.area : "", sep, "");
 		/* WARNING: depending upon implementations, snprintf may return
 		 * either -1 or the number of bytes that would be needed to store
 		 * the total message. In both cases, we must adjust it.
